### ISSUE TITLE
Add flexibility to "core site" definition in snippy-core?

### ISSUE THREAD

**Issue by amilesj** [Victorian-Bioinformatics-Consortium/snippy#162]
I'm not sure how much of a pain this would be, but it would be very cool if future versions of snippy allowed the user to modify the "core site" threshold from being present in 100% of samples to being present in 99% of samples, 95% of samples, etc.

**Comment by tseemann** [comment]
@amilesj it is something that has been requested before.
May i ask why you want those?
Is it because some of your samples aren't as good quality?

And if you have an alignment, what will you do with the `N` and `-` bases?
Most tree software will ignore those columns anyway.

**Comment by amilesj** [comment]
Hi Torsten, sorry for not responding to your questions sooner.

Of course! It is just a matter of losing information/resolution. It is my understanding that likelihood-based phylogenetic programs **do** utilize information in sites where some samples have `-` or `N` characters, rather than dropping the site entirely. Joe Felsenstein's _Inferring Phylogenies_ book is cited for the reasoning behind this, and the developers of RAxML and BEAST reference it [here](https://groups.google.com/forum/#!searchin/raxml/gap$20missing%7Csort:date/raxml/kry0iytH2Tg/h5XTol6dCAAJ), [here](https://groups.google.com/forum/#!searchin/raxml/treats$201%7Csort:date/raxml/c42B-rIXRVM/jjuH292dCAAJ), and [here](https://groups.google.com/forum/#!topic/beast-users/ixrGUA1p4OM)

An example from my data: I used Kat Holt's `parseSNPtable.py` script from the RedDog pipeline to play with the cutoffs. This data consists of 128 very closely related (sub-sequence type level) E. coli. After filtering phage and recombination-associated sites, if I use a 100% core gene requirement, I have 2875 sites in my alignment. If I drop that to 99% or 95%, I have 3181 or 3347 sites in my alignment, respectively. These differences aren't big enough to change the tree topology, but they do affect branch lengths. And with such closely related strains, it could be enough to make two samples seem identical where there are differences. 

Lower quality samples could definitely contribute to this...I need to dig more into my data to see if that is the case, although I would be surprised if this is driving the differences since I did a good amount of quality checks.

**Comment by amilesj** [comment]
Yes, I took the `.nway.tab` file output from Snippy and converted it to a `.csv` file that the `parseSNPtable.py` script from Red Dog would read. 

Yes, I see what you mean. As you said, some tools (`RAxML` included) do treat the `N` and `-` characters the same, mathematically. I'm afraid the implications of this are beyond my phylogenetics expertise! 

You are welcome. Thanks for considering it!

**Comment by eam12** [comment]
I just came across this thread and agree that allowing for a non-strict core would be a great feature to add to Snippy. I'm also dealing with very closely related strains (<1000 SNPs) and would benefit from the added resolution of slightly decreasing the "core site" threshold. Thank you!
